lib/src/sai_redis_notifications.cpp

```diff
@@ -216,7 +216,11 @@ void handle_notification(
     }
     else if (notification == "fdb_event")
     {
-        handle_fdb_event(data);
+        /* The fdb handling is moved to fdbOrch so that both
+         * reference count updates (sai_redis and portsOrch)
+         * happen in same context to avoid any mismatch.
+         */
+        /* handle_fdb_event(data);*/
```

Comment:
no no no, this must be handled here, its counting references and all other stuff, please revert

Reply:
> no no no, this must be handled here, its counting references and all other stuff, please revert

This function call has been moved to fdborch and is called from the orchagent main thread when fdb event is received. It was needed to avoid inconsistent reference count between fdborch and sai_redis, as now we maintain a per vlan and per port reference count in portsorch/fdborch which can be referenced before deleting bridge port or vlan in ASIC_DB.
Is it absolutely necessary to call this function here from the notification thread? There is no change in functionality except that it is being called from a different thread.

If it is absolutely needed to call this function from notification thread, then I plan to add additional data structure in fdborch to maintain the reference count instead of using the existing m_portlist and then call fdborch fdb event handling from on_fdb_event() callback, which is invoked  by sai_redis notification thread. 
This approach will also require a new semaphore for access to this new data structure as it will be updated by both notification thread and orchagent main thread (the main thread will update it when bridge port/vlan is added/deleted and also it will check the reference count before add/delete bridge port/vlan from ASIC_DB).

Let me know whether you agree with this approach.

Reply:
i get that, but it cant be moved to orchagent, since sairedis must be selfcontaing library, cant be splitted apart,  please revert

Reply:
> i get that, but it cant be moved to orchagent, since sairedis must be selfcontaing library, cant be splitted apart, please revert

Done